mobile/lib/ui/viewer/actions/album_selection_action_widget.dart

```diff
@@ -0,0 +1,171 @@
+        onTap: _onPinClick,
+      ),
+    );
+    items.add(
+      SelectionActionButton(
+        labelText: S.of(context).delete,
+        icon: Icons.delete_outline,
+        onTap: _trashCollection,
+      ),
+    );
+    items.add(
+      SelectionActionButton(
+        labelText: S.of(context).hide,
+        icon: Icons.visibility_off_outlined,
+        onTap: _onHideClick,
+      ),
+    );
+    final scrollController = ScrollController();
+
+    return MediaQuery(
+      data: MediaQuery.of(context).removePadding(removeBottom: true),
+      child: SafeArea(
+        child: Scrollbar(
+          radius: const Radius.circular(1),
+          thickness: 2,
+          controller: scrollController,
+          thumbVisibility: true,
+          child: SingleChildScrollView(
+            physics: const BouncingScrollPhysics(
+              decelerationRate: ScrollDecelerationRate.fast,
+            ),
+            scrollDirection: Axis.horizontal,
+            child: Container(
+              padding: const EdgeInsets.only(bottom: 24),
+              child: Row(
+                crossAxisAlignment: CrossAxisAlignment.start,
+                children: [
+                  const SizedBox(width: 4),
+                  ...items,
+                  const SizedBox(width: 4),
+                ],
+              ),
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+
+  Future<void> _shareCollection() async {
+    await collectionActions.shareMultipleCollectionSheet(
+      context,
+      widget.selectedAlbums.albums.toList(),
+    );
+    widget.selectedAlbums.clearAll();
+  }
+
+  Future<void> _trashCollection() async {
+    int count = 0;
+    final List<Collection> nonEmptyCollection = [];
+
+    for (final collection in widget.selectedAlbums.albums) {
+      count = await FilesDB.instance.collectionFileCount(collection.id);
+      final bool isEmptyCollection = count == 0;
+      if (isEmptyCollection) {
+        try {
+          await CollectionsService.instance.trashEmptyCollection(collection);
+        } catch (e, s) {
+          _logger.warning("failed to trash collection", e, s);
+          await showGenericErrorDialog(context: context, error: e);
+        }
+      } else {
+        nonEmptyCollection.add(collection);
+      }
+    }
+
+    if (nonEmptyCollection.isNotEmpty) {
+      final bool result = await collectionActions.deleteMultipleCollectionSheet(
+        context,
+        nonEmptyCollection,
+      );
+      if (result == true) {
+        Navigator.of(context).pop();
+      } else {
+        debugPrint("No pop");
+      }
+    }
+    widget.selectedAlbums.clearAll();
+  }
+
+  Future<void> _onPinClick() async {
+    for (final collection in widget.selectedAlbums.albums) {
+      if (collection.isPinned) continue;
+      await updateOrder(
+        context,
+        collection,
+        collection.isPinned ? 1 : 1,
+      );
+    }
+    widget.selectedAlbums.clearAll();
+  }
+
+  Future<void> _onHideClick() async {
```

Comment:
Certain actions like Hide or Pinning/Trash are only supported on albums that are owned by the user.
Also, we don't support trashing/hiding/pinning fav collections. Have we tested for these flows?